python/pyarrow/dataset.py

```diff
@@ -155,46 +155,55 @@ def partitioning(schema=None, field_names=None, flavor=None,
 
     Specify the Schema for paths like "/2009/June":
 
-    >>> partitioning(pa.schema([("year", pa.int16()), ("month", pa.string())]))
+    >>> import pyarrow as pa
+    >>> import pyarrow.dataset as ds
+    >>> ds.partitioning(pa.schema([("year", pa.int16()),
+    ...                            ("month", pa.string())]))
+    <pyarrow._dataset.DirectoryPartitioning object at ...>
```

Comment:
How useful is it to see those outputs? (we could try to make the repr more informative though) 
Because if we think that is not useful, we could also do something like `part = ..` to not have to add the output.